tests_e2e/tests/hibernation/hibernation.py

```diff
@@ -0,0 +1,214 @@
+        for record in goal_states:
+            if "source: Fabric" in record.message:
+                raise Exception(f"A Fabric goal state occurred after hibernation. This invalidates the test results. Goal state: {record.text}")
+        log.info("The agent processed only FastTrack goal states after hibernation...")
+
+    def _get_tenant_certificate(self) -> str:
+        stdout = self._ssh_client.run_command("get_goal_state.py --certificates --expand", use_sudo=True)
+        candidates = [line for line in stdout.splitlines() if line.endswith(".prv")]  # Only the tenant certificate should include a private key
+        if len(candidates) != 1:
+            raise Exception(f"Could not find the tenant certificate. Current certificates: {stdout}")
+        return os.path.basename(candidates[0]).replace('.prv', '')
+
+    def _enable_hibernation(self) -> None:
+        #
+        # The test may be running on an existing machine where hibernation has already been set as an additional capability, so we check that before setting it.
+        #
+        log.info("Enabling hibernation on %s", self._context.vm)
+        model: VirtualMachine = self._context.vm.get_model(include_instance_view=True)
+        if model.additional_capabilities is not None and model.additional_capabilities.hibernation_enabled:
+            log.info("Hibernation is already an additional capability of %s", self._context.vm)
+        else:
+            #
+            # To enable hibernation, the machine must be deallocated, then the hibernationEnabled property must be set, and, lastly, the machine must be reallocated
+            #
+            log.info("Deallocating %s in order to enable hibernation", self._context.vm)
+            self._context.vm.deallocate()
+            log.info("Adding hibernation as an additional capability to %s", self._context.vm)
+            self._context.vm.update(
+                {
+                    "additionalCapabilities": {
+                        "hibernationEnabled": True
+                    }
+                }
+            )
+            log.info("Reallocating %s after enabling hibernation", self._context.vm)
+            self._context.vm.start()
+            self._refresh_ip_address()
+
+        #
+        # The test may be running on an existing machine where the Hibernate extension has already been installed, so we check that before installing it.
+        #
+        if model.instance_view.extensions is not None and any(e for e in model.instance_view.extensions if e.type == "Microsoft.CPlat.Core.LinuxHibernateExtension"):
+            log.info("The Hibernation extension is already installed")
+        else:
+            log.info("Installing the Hibernation extension")
+            hibernate = VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.Hibernate, resource_name="Hibernate")
+            hibernate.enable(auto_upgrade_minor_version=True)
```

Comment:
suggestion: add `log.info("")` at the end of this method. there's lots of logs as a part of enabling hibernation

Reply:
Agree, but I added the blank line on the main test method instead. Thanks for the suggestions